### DISCUSSION TITLE
building logic latch in the micro-controller Program

### DISCUSSION THREAD

**Discussion by zafarmehdi** [espressif/arduino-esp32#7366] [Q&A]
![latch logic](https://user-images.githubusercontent.com/114767838/196278348-728b04d9-7745-458e-aadc-1dd2b7f64d63.PNG)
I want to build this latch in Arduino / ESP32 / ESP8266 through programming.
this is a latch wherein two switches are used to control a water pump. when the tank goes empty (both inputs go low) the pump starts as the pump start pouring water the low-level input goes high but the output will not change and the pump will continue to run.  When the low and high both inputs go high the output goes low and the pump stops pouring water but did not goes high on going high-level input to low until both inputs go low.

**Comment by mrengineer7777** [comment]
This can be implemented with a simple series of "if" statements.

```
uint8_t input1 = 0;
uint8_t input2 = 0;

void loop(void) {
   //Read switch states into input. Use a sub-function to debounce inputs.

   if(input1==0 && input2==0) {
      //Motor = running
   }
   else if(input1==1 && input2==1) {
      //Motor = stopped
   }
}
```
